internal/archiver/archiver.go

```diff
@@ -819,7 +825,7 @@ func (arch *Archiver) Snapshot(ctx context.Context, targets []string, opts Snaps
 		return nil, restic.ID{}, err
 	}
 
-	sn, err := restic.NewSnapshot(targets, opts.Tags, opts.Hostname, opts.Time)
+	sn, err := restic.NewSnapshot(cleanPaths, opts.Tags, opts.Hostname, opts.Time)
```

Comment:
I think this could have an impact on parent snapshot detection.

[Here](https://github.com/restic/restic/pull/3200/files#diff-30d8672eafa9c0585766b6f437530377563cd1aac5606674c366f3ede4abea9eR596) we use `targets` provided by the user to locate the parent snapshot. On the current line we use `cleanPaths` for the snapshot paths. If `resolveRelativeTargets` encounters a relative target (like `.`) then these two values will be different, and a subsequent backup would fail to locate the parent snapshot.

Is there any reason not to use `opts.Paths` directly?

Reply:
Good catch! Is changed now!